### PR TITLE
Feature/webpack fixes

### PR DESCRIPTION
**Pull request by Harvel218**

### CODE REVIEW

File: src/Resources/assets/admin/js/bitbag/bitbag-page-slug.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+            if (!textField || !slugField || !locale) {
+                return;
+            }
+
+            let timeout;
+
+            textField.addEventListener('input', (e) => {
+                e.preventDefault();
+                if (slugField.readOnly) {
```
Comment (tomaszsiwarski) [inline src/Resources/assets/admin/js/bitbag/bitbag-page-slug.js:49]: always add newline before and after if statement (if there is some code before it/after it with the same number of tabs)

File: src/Resources/views/Block/Crud/_form.html.twig [hunk 2]
```diff
@@ -14,7 +14,12 @@
             {{ form_row(form.sections) }}
             {{ form_row(form.channels) }}
 
-            <a href="#" class="ui labeled button icon primary bitbag-cms-resource-preview" data-url="{{ path('bitbag_sylius_cms_plugin_admin_block_preview') }}">
+            <a 
+                href="#" 
+                class="ui labeled button icon primary bitbag-cms-resource-preview"  
+                data-url="{{ path('bitbag_sylius_cms_plugin_admin_block_preview') }}"
+                data-bb-cms-preview-btn
+                >
```
Comment (tomaszsiwarski) [inline src/Resources/views/Block/Crud/_form.html.twig:22]: tabs number should match `<a`

File: tests/Application/.gitignore [hunk 3]
```diff
@@ -20,3 +21,5 @@
 ###> symfony/web-server-bundle ###
 /.web-server-pid
 ###< symfony/web-server-bundle ###
+
```
Comment (tomaszsiwarski) [inline tests/Application/.gitignore:24]: ?

File: tests/Application/assets/shop/entry.js [hunk 4]
```diff
@@ -0,0 +1 @@
+import 'sylius/bundle/ShopBundle/Resources/private/entry';
```
Comment (tomaszsiwarski) [inline tests/Application/assets/shop/entry.js:1]: ?

File: tests/Application/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+const adminConfig = Encore.getWebpackConfig();
+
+adminConfig.resolve.alias['sylius/ui'] = uiBundleScripts;
+adminConfig.resolve.alias['sylius/ui-resources'] = uiBundleResources;
+adminConfig.resolve.alias['sylius/bundle'] = syliusBundles;
+adminConfig.externals = Object.assign({}, adminConfig.externals, { window: 'window', document: 'document' });
+adminConfig.name = 'admin';
+
+module.exports = [shopConfig, adminConfig, bitbagCmsShop, bitbagCmsAdmin];
```
Comment (tomaszsiwarski) [inline tests/Application/webpack.config.js:48]: ?

File: src/Resources/assets/admin/js/bitbag/bitbag-cms-preview.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        const form = this.button.closest('form');
+
+        const settings = {
+            method: 'POST',
+            body: new FormData(form),
+        };
+
+        try {
+            triggerCustomEvent(this.mediaContainer, 'cms.create.preview.start');
+            const req = await fetch(`${path}?_channel_code=${channelCode}&_locale=${localeCode}`, settings);
```
[the remark this answers is not in the document]
Comment (Harvel218) [reply src/Resources/assets/admin/js/bitbag/bitbag-cms-preview.js:74]: its width is around 80 - the breakpoint in prettier is at 120, so there is still a lot of free space, i would leave it as is.

File: tests/Application/assets/shop/entry.js (hunk 4 above)
Comment (Harvel218) [reply tests/Application/assets/shop/entry.js:1]: its required for sylius assets in test environment

File: tests/Application/webpack.config.js (hunk 5 above)
Comment (Harvel218) [reply tests/Application/webpack.config.js:48]: shop/admin config - its base sylius styles for test env, shop/admin - cms is the CMS plugin custom styles that need to be applied.

Config for cms plugin is imported from root and used in test/app webpack, then you can export configs/entries and use them in other places.

Comment (tomaszsiwarski) [reply tests/Application/webpack.config.js:48]: I only meant missing newline xD

File: tests/Application/assets/shop/entry.js (hunk 4 above)
Comment (tomaszsiwarski) [reply tests/Application/assets/shop/entry.js:1]: I only meant missing newline xD

**Review by tomaszsiwarski — APPROVED** [review]

**Comment by Prometee** [comment]
@tomaszsiwarski @Harvel218 @KrisBitBag do you plan to add an UPGRADE note about this.
 I mean it could be cool to know how to migrate from gulp to webpack config.

**Comment by Harvel218** [comment]
> @tomaszsiwarski @Harvel218 @KrisBitBag do you plan to add an UPGRADE note about this. I mean it could be cool to know how to migrate from gulp to webpack config.

Yes, it will be done soon (few days from now) - whole documentation update.

**Comment by Prometee** [comment]
@Harvel218 I manage to configure my own project but I don't do it the same way you did, I didn't want multiple CSS or JS generation, so I simply add an import inside each of my `entry.js` file like this :

Inside `assets/admin/entry.js` :

```js
...

import '../../vendor/bitbag/cms-plugin/src/Resources/assets/admin/entry';
```

Inside `assets/shop/entry.js` :

```js
...

import '../../vendor/bitbag/cms-plugin/src/Resources/assets/shop/entry';
```

And I have to update my `package.json` with the new packages versions you changed :

```json
...

    "devDependencies": {
        "@symfony/webpack-encore": "^1.6.1",
        ...
        "node-sass": "^6.0.1",
        "sass-loader": "^12.3.0",
    }
```

**Comment by Harvel218** [comment]
@Prometee Its also a way to do this, there will be like 5 or 6 options, describing how You can manage plugin assets (I mentioned like 2 or 3 of them in sylius slack with You), Im happy you managed that!

**Pull request merged**